core/src/main/java/org/elasticsearch/search/profile/SearchProfileShardResults.java

```diff
@@ -75,26 +87,71 @@ public void writeTo(StreamOutput out) throws IOException {
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject("profile").startArray("shards");
-
-        for (Map.Entry<String, ProfileShardResult> entry : shardResults.entrySet()) {
+        builder.startObject(PROFILE_FIELD).startArray(SHARDS_FIELD);
+        TreeSet<String> sortedKeys = new TreeSet<>(shardResults.keySet());
```

Comment:
Are we forced to sort all shard results every time just to be able to check them in tests? Can't we check / sort again in the test instead?